src/Compiler/Microsoft.AspNetCore.Mvc.Razor.Extensions/test/RazorPageDocumentClassifierPassTest.cs

```diff
@@ -19,7 +19,14 @@ public class RazorPageDocumentClassifierPassTest : RazorProjectEngineTestBase
     public void RazorPageDocumentClassifierPass_LogsErrorForImportedPageDirectives()
     {
         // Arrange
-        var sourceSpan = new SourceSpan("import.cshtml", 0, 0, 0, 5);
+        var sourceSpan = new SourceSpan(
+            filePath: "import.cshtml",
+            absoluteIndex: 0,
+            lineIndex: 0,
+            characterIndex: 0,
+            length: 5,
+            lineCount: 0,
+            endCharacterIndex: 5);
```

Comment:
I don't understand why `SourceSpan`s have more arguments now in tests.

Reply:
Previously, diagnostics would be considered equal, even if they had different `lineCount` or `endCharacterIndex` values. Now that their equality is based on checksums which include all SourceSpan values, this isn't true. Essentially, `SourceSpan` was updated with new properties at some point, but the `GetHashCode` and `Equals` methods weren't updated. This is something that needs to be fixed, because it's almost certainly a source of subtle bugs.